### ISSUE TITLE
RunningAverageAlgorithm Issues with Stores during DST for ~1Day TIme Series

### ISSUE THREAD

**Issue by jbatterman** [opendcs/opendcs#562]
**Describe the bug**
MRBWM is having an issue getting a regular-interval time series to store in OpenDCS.  If the output time series for the RunningAverageAlgorithm is a LRTS, data will not store during DST (i.e., stores at 24:00 outside of DST and does not store during DST).  If the output time series is a pseudo-regular time series, data will store but at shifted intervals (i.e., store at 24:00 outside of DST and at 01:00 during DST).

**To Reproduce**
1. Delete the existing *.*.*.~1Day.3Days.Rev-NWO-Evap time series
2. Recreating in OpenDCS only creates them as pseudo-regular time series.  To create as LRTS, create in CWMSVue or use SQL Developer to change the time series offset (from large negative number to 0).
3. To test psudo-regular, wanting to store at 24:00 hrs all year, click Run Computations and test over the DST change (back to March 1)
4. Test with aggregateTimeZone property of US/Central or as UTC with the same results.  Run Computations will show all output data at the same hour but does not store that way.
5. To test LRTS, change time series or create as LRTS and then click Run Computations.  Data will all show at the same hour, but will only store in Oracle up to DST.

**Expected behavior**
LRTS should store one value a day and not shift during DST.  Expected output would be at 24:00 hrs all year.  Other computations that are running with pseudo-regular time series are showing output at 24:00 all year, but this particular algorithm is not.  Examples that store correctly for pseudo-regular are AverageAlgorithm and SubSample.  For MRBWM using LRTS, daily data is stored using a script and it successfully stores at 24:00 all year.

**Desktop/ServerSide (please complete the following information):**
 - OS: Windows 10, version 7.0.1
 - Server schema 6.8.14

**Additional context**
This behavior is resulting in difficulty implementing the Evap time series into MRBWM daily operations, which use scripts to calculate daily inflows and balance reservoirs on a monthly basis.  MRBWM will meet with @MikeNeilson and @zamusacelre on May 7 regarding this issue.

**Comment by MikeNeilson** [comment]
Is the process itself running at UTC?

**Comment by jbatterman** [comment]
It's running on the "testing" computation process.  Not sure how to tell what time zone the process is running - I believe it must be UTC since that last upgrade required our shell to be switched to UTC?

**Comment by MikeNeilson** [comment]
Yeah, it should then. that would've been that default TZ set in the user.properties file.

**Comment by jbatterman** [comment]
Ok yes - we updated our user.properties file to use UTC on the last upgrade.

**Comment by jbatterman** [comment]
After looking at this, we determined that the RunningAverageAlgorithm is determining its time zone offset using the input role rather than the output (average) role.  In this case the computation is using a regular interval 1.Day1.Day time series as input and looking for a ~1Day.3Day output, so the input time series is resulting in the storing offset during DST.  Other working computations/algorithms are using sub-daily interval data (i.e., 1Hour) to calculate ~1Day data, and that is correctly calculating and storing.
